app/scripts/directives/mobileAssociations.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileAssociations', {
+    controller: [
+      MobileAssociationsCtrl
+    ],
+    bindings: {
+      associated: '<',
+      excluded: '<',
+      exclude: '<',
+      associate: '<',
+      serviceClasses: '<',
+      heading: '<',
+      type: '<'
+    },
+    templateUrl: 'views/directives/mobile-associations.html'
+  });
+
+  var clientAssociation = function(mobileResource) {
+    return {
+      uid: _.get(mobileResource, 'metadata.uid'),
+      id: _.get(mobileResource, 'spec.appIdentifier'),
+      name: _.get(mobileResource, 'spec.name'),
+      icon: _.get(mobileResource, 'metadata.annotations.icon'),
+      object: mobileResource
+    };
+  };
+
+  var serviceAssociation = function(serviceResource, serviceClass) {
+    return {
+      uid: _.get(serviceResource, 'metadata.uid'),
+      id: _.get(serviceResource, 'metadata.name'),
+      name: _.get(serviceResource, 'metadata.labels.serviceName'),
+      imageUrl: _.get(serviceClass, 'spec.externalMetadata.imageUrl'),
+      object: serviceResource
+    };
+  };
+
+  function MobileAssociationsCtrl() {
+    var ctrl = this;
+
+    ctrl.$onChanges = function(changes) {
+      if (changes.excluded) {
+        ctrl.hasExcluded = !_.isEmpty(ctrl.excluded);
+        ctrl.excludedResources = ctrl.createAssociations(ctrl.excluded);
+      }
+
+      if (changes.associated) {
+        ctrl.associatedResources = ctrl.createAssociations(ctrl.associated);
+      }
+    };
+
+    ctrl.createAssociations = function(resources) {
+      return _.map(resources, function(resource) {
+        if (resource.kind === 'MobileClient') {
```

Comment:
Do you think this value here will be used by others in the future? Will it be worth to have it as a constant?

Reply:
yeah I checked in the codebase and this type of check occurs a lot and they are just using strings checks like this. I'll leave it as is until a situation comes up where we need a constant